project-base/storefront/pages/products/[productSlug].tsx

```diff
@@ -1,58 +1,28 @@
-import { ProductDetailContent } from 'components/Pages/ProductDetail/ProductDetailContent';
-import { ProductDetailMainVariantContent } from 'components/Pages/ProductDetail/ProductDetailMainVariantContent';
 import {
-    useProductDetailQuery,
     TypeProductDetailQuery,
     TypeProductDetailQueryVariables,
     ProductDetailQueryDocument,
 } from 'graphql/requests/products/queries/ProductDetailQuery.generated';
-import { useGtmFriendlyPageViewEvent } from 'gtm/factories/useGtmFriendlyPageViewEvent';
-import { useGtmPageViewEvent } from 'gtm/utils/pageViewEvents/useGtmPageViewEvent';
 import { NextPage } from 'next';
-import { useRouter } from 'next/router';
+import dynamic from 'next/dynamic';
+import { Suspense } from 'react';
 import { OperationResult } from 'urql';
 import { createClient } from 'urql/createClient';
 import { handleServerSideErrorResponseForFriendlyUrls } from 'utils/errors/handleServerSideErrorResponseForFriendlyUrls';
 import { isRedirectedFromSsr } from 'utils/isRedirectedFromSsr';
 import { getSlugFromServerSideUrl } from 'utils/parsing/getSlugFromServerSideUrl';
-import { getSlugFromUrl } from 'utils/parsing/getSlugFromUrl';
 import { getServerSidePropsWrapper } from 'utils/serverSide/getServerSidePropsWrapper';
 import { ServerSidePropsType, initServerSideProps } from 'utils/serverSide/initServerSideProps';
 
-const ProductDetailPage: NextPage<ServerSidePropsType> = () => {
-    const router = useRouter();
-    const [{ data: productData, fetching }] = useProductDetailQuery({
-        variables: { urlSlug: getSlugFromUrl(router.asPath) },
-    });
-
-    const product =
-        productData?.product?.__typename === 'RegularProduct' || productData?.product?.__typename === 'MainVariant'
-            ? productData.product
-            : null;
-
-    const pageViewEvent = useGtmFriendlyPageViewEvent(product);
-    useGtmPageViewEvent(pageViewEvent, fetching);
+const ProductDetailWrapper = dynamic(() =>
+    import('components/Pages/ProductDetail/ProductDetailWrapper').then((component) => component.ProductDetailWrapper),
+);
 
+const ProductDetailPage: NextPage<ServerSidePropsType> = () => {
     return (
-        <CommonLayout
-            breadcrumbs={product?.breadcrumb}
-            breadcrumbsType="category"
-            canonicalQueryParams={[]}
-            description={product?.seoMetaDescription}
-            hreflangLinks={product?.hreflangLinks}
-            isFetchingData={fetching}
-            title={product?.seoTitle || product?.name}
-        >
-            {product?.__typename === 'RegularProduct' && <ProductDetailContent fetching={fetching} product={product} />}
-
-            {product?.__typename === 'MainVariant' && (
-                <ProductDetailMainVariantContent fetching={fetching} product={product} />
-            )}
-
-            <LastVisitedProducts currentProductCatnum={product?.catalogNumber} />
-        </CommonLayout>
+        <Suspense>
```

Comment:
Hmm, so here Suspense works as we want? 🤔

Reply:
Short answer, yes. This has proved to be very helpful. Long answer where I give you reasons why it works would be really hard. Though I understand how Suspense works, here I can only give an intuitive answer.

Suspense works as a boundary (same as ErrorBoundary, but for `throw Promise` instead of `throw Error`). It takes the hydration and defers it to a later stage. However, our code works in a way that it has very little effect (we do not throw that many processes). Why it works here is (I suppose) because of those deferred components. Since there is nothing to be hydrated, the hydration task disappears completely when used. 

With that being said, it only works if I use both. If I only use Suspense + dynamic import, it has no effect. If I only use timeouts, it has some effect. When I use both, it has a superboosted effect.